src/components/form/HdRadioCard.vue

```diff
@@ -0,0 +1,280 @@
+import deepmerge from 'deepmerge';
+import { getMessages } from 'homeday-blocks/src/lang';
+import formFieldMixin from './formFieldMixin';
+import HdRadioIndicator from './HdRadioIndicator.vue';
+
+export default {
+  name: 'HdRadioCard',
+  mixins: [formFieldMixin],
+  components: {
+    HdRadioIndicator,
+  },
+  props: {
+    name: {
+      type: String,
+      required: true,
+    },
+    value: [String, Number],
+    nativeValue: {
+      type: [String, Number],
+      required: true,
+    },
+    ariaLabel: String,
+    ariaLabelledBy: String,
+    disabled: Boolean,
+    required: Boolean,
+    isInvalid: {
+      type: Boolean,
+      default: false,
+    },
+    mode: {
+      type: String,
+      default: 'card',
+      validator: (value) => ['card', 'tile'].includes(value),
+    },
+    texts: {
+      type: Object,
+      default: () => ({}),
+    },
+    lang: {
+      type: String,
+      default: 'de',
+    },
+    /** @type {import('vue').PropOptions<{ validate: (value: unknown) => boolean, errorMessage: string }[]>} */
+    customRules: {
+      type: Array,
+      default: () => [],
+      validator: (rulesProvided) => rulesProvided.every(
+        ({ validate, errorMessage }) => typeof validate === 'function' && typeof errorMessage === 'string',
+      ),
+    },
+  },
+  data() {
+    return {
+    /** @type {string?} */
+      error: null,
+    };
+  },
+  watch: {
+    value() {
+      this.validate();
+    },
+    required() {
+      if (this.error) this.validate();
+    },
+    customRules() {
+      if (this.error) this.validate();
+    },
+  },
+  computed: {
+    /** @returns {import('homeday-blocks/src/lang').Messages} */
+    t() {
+      return deepmerge(getMessages(this.lang), this.texts);
+    },
+    /** @returns {boolean} */
+    isChecked() {
+      return this.nativeValue === this.value;
+    },
+    /** @returns {boolean} */
+    hasValidationErrors() {
+      return Boolean(this.error);
+    },
+    /** @returns {string} */
+    inputName() {
+      return `${this.name}-${this.nativeValue}`;
+    },
+  },
+  methods: {
+    validateForm() {
+      if (this.required && !this.value) return this.t.FORM.VALIDATION.REQUIRED;
+
+      const firstFailingRule = this.customRules.find(({ validate }) => !validate(this.value));
+      if (firstFailingRule) return firstFailingRule.errorMessage;
+
+      return null;
+    },
+    /**
+     * This is a default validate function
+     * it is a special function that is automatically
+     * invoked by HdForm.
+     *
+     * By surrounding HdRadioCard with a HdRadioCardGroup
+     * this function is automatically overriden.
+     *
+     * @returns {boolean}
+     */
+    validate() {
+      this.error = this.validateForm();
+      return !this.hasValidationErrors;
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+@import "homeday-blocks/src/styles/mixins.scss";
+
+::v-deep path {
+  transition: fill ($time-s * 2);
+}
+
+input[type="radio"] {
+  display: none;
```

Comment:
This would remove the input from the tab order. Is that what we want? Maybe we want to keep the tab order in place and just make it visually hidden?
We could even create a mixin for this:

```scss
/// Module export mixin
/// This mixin helps creating a practically hidden element. It's important to note that this
/// mixin will hide the element for the user, but will be kept available to screen readers
/// and other assistive technology users. They are hidden with CSS, but they remain on the DOM.
@mixin visually-hidden() {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0 0 0 0);
  border: 0;
}
```

Reply:
Well, this is not what we want, we already have the checkbox/radio surrounded by the label, if we use this snippet in this case, when the user is navigating through the form, they will have to hit the tab twice to select the next component.

Check the example below, I'm only navigating through the keyboard, and every tab I hit, it selects first the checkbox (that doesn't display any visual feedback), and when I hit again it selects the label.


https://user-images.githubusercontent.com/1855125/125812357-e29dbd1a-c5d0-4afc-ad57-67c5e97346ff.mov

What do you think about this?